bindings/java/src/main/com/apple/foundationdb/RangeResult.java

```diff
@@ -51,4 +51,22 @@
 		}
 		this.more = more;
 	}
+
+	RangeResult(DirectBufferIterator iterator) {
+		iterator.readResultsSummary();
+		more = iterator.hasMore();
+
+		int count = iterator.count();
+		values = new ArrayList<KeyValue>(count);
+
+		for (int i = 0; i < count; ++i) {
+			values.add(iterator.next());
+		}
+	}
+
+	public RangeResultSummary getSummary() {
```

Comment:
Given how this will always be called, it might make sense to either:

1. Create the `RangeResultSummary` at object construction time and memoize it for future calls.
1. Add methods to the `RangeResult` that can be used instead of the summary.

I suppose, this could also need to change if the API is changed to accommodate #3618, but that's a bridge that should probably be crossed when we get there.

Reply:
It is called just once per query, so I think we can skip for now, but yes, I can make the change if you think it is worth it atm.